src/ManageContainer/Client/ManageToDbGate/Client.go

```diff
@@ -322,55 +337,47 @@ func (c Client) getComputationResult(conn *grpc.ClientConn, jobUUID string) (*Co
 		return nil, err
 	}
 
-	var computationResults []ComputationResult
+	var computationResults []*ComputationResult
 	err = json.Unmarshal([]byte(response), &computationResults)
 	if err != nil {
 		return nil, err
 	}
 
-	if len(computationResults) != 1 {
+	if len(computationResults) == 0 {
 		return nil, errors.New("unique computation result could not be found: " + strconv.Itoa(len(computationResults)))
 	}
 
-	return &computationResults[0], nil
-}
-
-type Params struct {
-	JobUUID string      `json:"job_uuid"`
-	Result  interface{} `json:"result"`
+	return computationResults, nil
 }
 
 // DBGにモデルパラメータを送信する
-func (c Client) InsertModelParams(jobUUID string, params string) error {
+func (c Client) InsertModelParams(jobUUID string, params string, pieceId int32) error {
 	conn, err := connect()
 	if err != nil {
 		return err
 	}
 	defer conn.Close()
-	return c.insertModelParams(conn, jobUUID, params)
+	return c.insertModelParams(conn, jobUUID, params, pieceId)
 }
 
 // (conn)にモデルパラメータを送信する
-func (c Client) insertModelParams(conn *grpc.ClientConn, jobUUID string, params string) error {
-	cnt, err := c.count(conn, "job_uuid", jobUUID, "result")
-	ls.Lock(jobUUID)
-	defer ls.Unlock(jobUUID)
+func (c Client) insertModelParams(conn *grpc.ClientConn, jobUUID string, params string, pieceId int32) error {
+	// 重複チェック
+	exist, err := c.existPieceID(conn, "result", jobUUID, pieceId)
 	if err != nil {
 		return err
 	}
-	if cnt > 0 {
+	if exist {
 		return errors.New("重複データ登録エラー: " + jobUUID + "は既に登録されています．")
```

Comment:
@mdonaka 
It's more consistent this message in English.

```suggestion
		return errors.New("Duplicate error: " + jobUUID + " has already been registered.")
```

Reply:
Thanks for Suggested! I committed.